service/service.go

```diff
@@ -135,39 +120,10 @@ func dial(target string, cert *x509.Certificate, insecureAllowed bool) (*grpc.Cl
 	))
```

Comment:
check the logic of this function – `cert` is always nil, so we always run in insecure mode, which is not what we want – the bit before `grpc.Dial` should look something like

```
var tlsOpt grpc.DialOption
if insecureAllowed {
    tlsOpt = grpc.WithInsecure()
} else {
    creds := credentials.NewTLS(&tls.Config{})
    tlsOpt = grpc.WithTransportCredentials(creds)
}
```

…or possibly if you feel strongly about the implied semantics of "insecure*Allowed*" we should always `Dial` with transport credentials and – on suitable error – fall back to insecure mode